sentry_sdk/client.py

```diff
@@ -334,19 +335,27 @@ def capture_event(
         if session:
             self._update_session_from_event(session, event)
 
-        if event_opt.get("type") == "transaction":
-            # Transactions should go to the /envelope/ endpoint.
-            self.transport.capture_envelope(
-                Envelope(
-                    headers={
-                        "event_id": event_opt["event_id"],
-                        "sent_at": format_timestamp(datetime.utcnow()),
-                    },
-                    items=[
-                        Item(payload=PayloadRef(json=event_opt), type="transaction"),
-                    ],
-                )
+        attachments = hint.get("attachments")
+        is_transaction = event_opt.get("type") == "transaction"
+
+        if is_transaction or attachments:
+            # Transactions or events with attachments should go to the
```

Comment:
We're not trying to force upgrade to envelopes yet. There are many Sentry 9 installations out there for on prem that would break on the new SDK versions if we were to do that.

Reply:
Ah okay didn't know that. On Cocoa, we do that since 6.x.